src/vector/mod.rs

```diff
@@ -444,6 +464,25 @@ impl<T> Vector<T> {
     }
 }
 
+impl<T: Clone> Vector<T> {
+    /// Gets a mutable reference to an element. If the element is shared, this Vector will be cloned.
```

Comment:
I think it is best to say "If the element is shared it will be cloned.".  It is true for all mutable operations that *part* of the vector will be cloned.